src/windows/curve_editor.py

```diff
@@ -0,0 +1,657 @@
+
+        # Get data model and selection
+        model = get_app().window.propertyTableView.clip_properties_model.model
+        row = item.row()
+
+        property = model.item(row, 0).data()
+        property_type = property[1]["type"] # float, int, bool, color, string, reader
+        property_name = property[1]["name"] # localized value
+        property_key = property[0]
+
+        c = self.getClipByID(clip_id, item_type)
+
+        startTime = 0.0
+        endTime = 0.0
+        # Find parent clip to get the effect's correct starting position
+        if item_type == "effect":
+            effect = Effect.get(id=clip_id)
+            if not effect:
+                return
+            self.minTimeF = effect.parent["position"]
+            startTime = effect.parent["start"]
+            endTime = effect.parent["end"]
+        else:
+            self.minTimeF = c.data["position"]
+            startTime = c.data["start"]
+            endTime = c.data["end"]
+
+        # Start position of the curve has new offset now, recalculate
+        self.updateGridTimeShift()
+        self.curveOffsetTimeF = self.minTimeF - self.gridOffsetTime
+        
+        if c:
+            # Get clip attribute
+            if property_key in c.data:
+                # Determine type of keyframe (standard or color)
+                keyframe_list = []
+                if property_type == "color":
+                    keyframe_list = [c.data[property_key]["red"], c.data[property_key]["green"], c.data[property_key]["blue"]]
+                    color_key = ["red", "green", "blue"]
+                else:
+                    keyframe_list = [c.data[property_key]]
+                    color_key = None
+
+                # Total number of new curves to display
+                self.newCurNum = 0
+                
+                # Loop through each keyframe if any (red, blue, and green)
+                for i, keyframe in enumerate(keyframe_list):
+                    # Not all properties has Points member (not animated yet),
+                    # exclude file readers too - for example Source property of the Mask effect
+                    if type(c.data[property_key]) != dict or property_type == "reader":
+                        # Update info labels and abort graph building
+
+                        # The Effect object has no "title" property
+                        if item_type == "effect":
+                            obj_title = "name"
+                        else:
+                            obj_title = "title"
+
+                        # Get translation function
+                        _ = get_app()._tr
+                        self.crvEdt.setInfoLabels("{} ({})".format(c.data[obj_title], item_type), "{} ({})".format(property_name, _("No animation")))
+                        return
+
+                    # Number of animated points (keys of animation)
+                    self.newCurvPointsNum = len(keyframe["Points"])
```

Comment:
Hm ... `keyframe["Points"]` will access (in C++) `Keyframe::Points`, no? I'm about to remove this (public) member!

Reply:
You need to look at the `properties_model.py` code too then...

Reply:
Thanks for the hint, turned out to be no issue, `keyframe["Points"]` does *not* access (in C++) `Keyframe::Points` (almost everything seems to be passed back and forth using Json)